Bottom: c741adf8f27f09b86263a67cd51cb2b57a5bb96a
Top:    13390a9dd40f27965edc4ad8eed2098869d6efd9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-05 16:54:42 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 181ba6e..fd68900 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -235,6 +235,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
